clang/lib/Sema/SemaChecking.cpp

```diff
@@ -16134,7 +16134,7 @@ static void CheckConditionalOperator(Sema &S, AbstractConditionalOperator *E,
 /// Check conversion of given expression to boolean.
 /// Input argument E is a logical expression.
 static void CheckBoolLikeConversion(Sema &S, Expr *E, SourceLocation CC) {
-  if (S.getLangOpts().Bool)
+  if (S.getLangOpts().Bool && !S.getLangOpts().C23)
```

Comment:
I think it would be good to add some comments here as to why we're checking for !C23 explicitly, WDYT?

Reply:
Sounds very reasonable